core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -112,6 +118,9 @@ public static Environment prepareEnvironment(Settings input, Terminal terminal,
             throw new SettingsException("multiple settings files found with suffixes: "
                 + Strings.collectionToDelimitedString(foundSuffixes, ","));
         }
+        if (foundSuffixes.contains(".yml") || foundSuffixes.contains(".json")) {
+            deprecationLogger.deprecated("elasticsearch.yaml and elasticsearch.json are deprecated. Rename your configuration file to elasticsearch.yaml.");
```

Comment:
I think we can give a more specific message too: yml -> yaml only if yml, json -> yaml only if json instead of a general message covering both cases.